src/custom/state/ethFlow/updater.tsx

```diff
@@ -0,0 +1,33 @@
+import { Percent } from '@uniswap/sdk-core'
+import { useEffect, useState } from 'react'
+import { useSetUserSlippageTolerance, useUserSlippageTolerance } from 'state/user/hooks'
+import { useShowNativeEthFlowSlippageWarning } from 'state/ethFlow/hooks'
+
+export const ETH_FLOW_SLIPPAGE = new Percent(2, 100) // 2%
+
+export default function Updater() {
+  // use previous slippage for when user is not in native swap and set to native flow
+  const currentSlippage = useUserSlippageTolerance()
+  // save the last non eth-flow slippage amount to reset when user switches back to normal erc20 flow
+  const [erc20Slippage, setErc20Slippage] = useState<Percent | 'auto' | undefined>(currentSlippage)
+  // we show slippage warning when isNativeSwap and isUserNativeEthFlow are true
+  const shouldShowEthFlowSlippage = useShowNativeEthFlowSlippageWarning()
+  const setUserSlippageTolerance = useSetUserSlippageTolerance()
+  // change slippage set to 2% when detected eth swap and option is set to use native flow
+  useEffect(() => {
+    if (shouldShowEthFlowSlippage) {
+      // user switched back to erc20, we need to reset their previous slippage back to what it was
+      if (currentSlippage instanceof Percent && !currentSlippage.equalTo(ETH_FLOW_SLIPPAGE)) {
+        // do sth
+        setErc20Slippage(currentSlippage)
+      }
+      setUserSlippageTolerance(ETH_FLOW_SLIPPAGE)
```

Comment:
Why do we need this Updater? Could you explain why you like it better than this solution:

Defining a fn 
```
useGetSlippageTolerance() {
  const shouldShowEthFlowSlippage = useShowNativeEthFlowSlippageWarning()
  
shouldShowEthFlowSlippage
                    ? ETH_FLOW_SLIPPAGE.toSignificant(PERCENTAGE_PRECISION)
                    : slippageInput.length > 0
                    ? slippageInput
```
and then we use this function during slippage displays and during order placements?
Then we would never need to use:
```
      setUserSlippageTolerance(ETH_FLOW_SLIPPAGE)
```